src/index.tsx

```diff
@@ -20,122 +25,136 @@ export default class SignatureCanvas extends Component {
     clearOnResize: PropTypes.bool
   }
 
-  static defaultProps = {
+  static defaultProps: Pick<ReactSignatureCanvasProps, 'clearOnResize'> = {
     clearOnResize: true
   }
 
-  _sigPad = null
+  // this is some hack-ish init and casting to avoid `| null` everywhere :/
+  /* eslint-disable @typescript-eslint/consistent-type-assertions */
+  _sigPad: SignaturePad = {} as SignaturePad
+  _canvas: HTMLCanvasElement = {} as HTMLCanvasElement
+  /* eslint-enable @typescript-eslint/consistent-type-assertions */
+
+  _setRef = (ref: HTMLCanvasElement | null): void => {
+    if (ref !== null) {
+      this._canvas = ref
+    }
+  }
 
-  _excludeOurProps = () => {
+  _excludeOurProps = (): SignaturePad.SignaturePadOptions => {
     const { canvasProps, clearOnResize, ...sigPadProps } = this.props
     return sigPadProps
   }
 
-  componentDidMount () {
+  componentDidMount: Component['componentDidMount'] = () => {
     this._sigPad = new SignaturePad(this._canvas, this._excludeOurProps())
     this._resizeCanvas()
     this.on()
   }
 
-  componentWillUnmount () {
+  componentWillUnmount: Component['componentWillUnmount'] = () => {
     this.off()
   }
 
   // propagate prop updates to SignaturePad
-  componentDidUpdate () {
+  componentDidUpdate: Component['componentDidUpdate'] = () => {
     Object.assign(this._sigPad, this._excludeOurProps())
   }
 
   // return the canvas ref for operations like toDataURL
-  getCanvas = () => {
+  getCanvas = (): HTMLCanvasElement => {
     return this._canvas
   }
 
   // return a trimmed copy of the canvas
-  getTrimmedCanvas = () => {
+  getTrimmedCanvas = (): HTMLCanvasElement => {
     // copy the canvas
     const copy = document.createElement('canvas')
     copy.width = this._canvas.width
     copy.height = this._canvas.height
-    copy.getContext('2d').drawImage(this._canvas, 0, 0)
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    copy.getContext('2d')!.drawImage(this._canvas, 0, 0)
     // then trim it
     return trimCanvas(copy)
   }
 
   // return the internal SignaturePad reference
-  getSignaturePad = () => {
+  getSignaturePad = (): SignaturePad => {
```

Comment:
Something to think about in the future: exposing this as a public method has its drawbacks. User can execute:
```
ref.getSignaturePad().off()
```

which will be inconsistent with `off` implementation from `ReactSignatureCanvas` instance. Same goes for `on` method.

Reply:
The same can be said for `getCanvas()`. This is for advanced users so I assume they know what they're getting themselves into if they access the `signature_pad` instance.

It hasn't been an issue in any case, but if it were, I'd just put warnings in the docs as opposed to removing access to it (which would also be a breaking change).